ui/src/pages/backoffice/Acquisition/Order/OrderEditor/components/OrderForm/OrderForm.js

```diff
@@ -0,0 +1,193 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Header } from 'semantic-ui-react';
+import { getIn } from 'formik';
+import { SelectorField, TextField } from '@forms';
+import { Loader } from '@components';
+import { order as orderApi, vendor as vendorApi } from '@api/acquisition';
+import { AcquisitionRoutes } from '@routes/urls';
+import { goTo } from '@history';
+import {
+  BaseForm,
+  DateInputField,
+  GroupField,
+  PriceField,
+  SelectField,
+  StringField,
+} from '@forms';
+import { serializeVendor } from '@components/ESSelector/serializer';
+import { Funds, OrderLines, Payment } from './components';
+import { vocabulary as vocabularyApi } from '@api';
+import { invenioConfig } from '@config';
+import { sessionManager } from '@authentication/services';
+
+const orderSubmitSerializer = (values, newRecord) => {
+  const submitValues = { ...values };
+  if (newRecord) {
+    submitValues.created_by_pid = sessionManager.user.id;
+  }
+  submitValues.vendor_pid = values.vendor.pid;
+  return submitValues;
+};
+
+export class OrderForm extends Component {
+  state = {
+    isLoading: true,
+    error: null,
+    currencies: [],
+  };
+
+  componentDidMount() {
+    this.fetchCurrencies();
+  }
+
+  query = () => {
+    const searchQuery = vocabularyApi
+      .query()
+      .withType(invenioConfig.vocabularies.order.acq_currency)
+      .qs();
+    return vocabularyApi.list(searchQuery);
+  };
+
+  serializer = hit => ({
+    key: hit.metadata.key,
+    value: hit.metadata.key,
+    text: hit.metadata.key,
```

Comment:
```suggestion
    text: hit.metadata.text,
```
Would be easier for the user to read, what do you think?